mlcluster.template

```diff
@@ -383,7 +407,7 @@ Resources:
         S3Directory: !FindInMap [Variable,"S3Directory","base"]
         NodesPerZone: !Ref NodesPerZone
         NumberOfZones: !Ref NumberOfZones
-        Subnets: !If [MultiZone, !Join [',', !Ref PrivateSubnets], !Select [0, !Ref PrivateSubnets]]
+        Subnets: !If [MultiZone, !Join [',', [!Ref PrivateSubnet1, !Ref PrivateSubnet2, !Ref PrivateSubnet3]], !Ref PrivateSubnet1]
```

Comment:
Does Fn::Join work with empty value? Since subnet2 and 3 may be empty.

Thanks,
Vanessa

Reply:
No.

Reply:
Yes, the Join won't work for the empty values. This is why we have the MultiZone condition. This condition is true when we provide 3 AZs. So as the mapping goes, when multizone is true we use join. If it is false, then we use the First private/public subnet

MultiZone:
    !Not [!Equals [!Ref NumberOfZones, 1]]